File: HR_Management_backend/src/main/java/com/hrManagement/service/EmployeeService.java
package com.hrManagement.service;

import com.hrManagement.dto.EmployeeDTO;
import com.hrManagement.dto.UserLoginDTO;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface EmployeeService {

    EmployeeDTO addEmployee(EmployeeDTO employeeDTO);

    List<EmployeeDTO> getAllEmployees();

    Boolean deleteEmployee(String employeeId);

    ResponseEntity<UserLoginDTO> userLogin(UserLoginDTO userLoginDto);
}

File: hr_management_frontend/src/app/hrManagement/services/employee.service.ts

import { Injectable } from '@angular/core';
import {ApiService} from "./api.service";
import {HttpClient, HttpParams} from "@angular/common/http";
import {Observable} from "rxjs";
import {DepartmentDto} from "../dto/DepartmentDto";
import {EmployeeDto} from "../dto/EmployeeDto";

@Injectable({
  providedIn: 'root'
})
export class EmployeeService extends ApiService{
  constructor(private http: HttpClient) {
    super();
  }

  addEmployee(employeeDto: EmployeeDto) {
    return this.http.post<EmployeeDto>(this.MAIN_URL + this.URL + '/addEmployee', employeeDto);
  }
  getAllEmployees(): Observable<Array<EmployeeDto>> {
    return this.http.get<Array<EmployeeDto>>(this.MAIN_URL + this.URL + '/getAllEmployees');
  }
  deleteEmployee(employeeId: any): Observable<boolean> {
    const params = new HttpParams()
      .set('employeeId', employeeId);
    return this.http.post<boolean>(this.MAIN_URL + this.URL + '/deleteEmployee', params);
  }


}
